### PR TITLE
feat: Support using Sinco in docker

### LINKED ISSUE #46: Add Support for Docker
## Summary

What:

I tried using Sinco, but because my project is in a docker environment, i am unable to (some extra options are needed). This is to add support for docker

Why:

Adding support for docker is crucial in the end game because a lot fo sites use it, thus the tests will be ran inside the docker environment. 

Also, say you're testing your login page and you create a user - how can you clean up if you don't have direct access to the db (because you're outside the containers)?

## Acceptance Criteria

Below is a list of tasks that must be completed before this issue can be closed.

- [ ] Write documentation
- [ ] Write unit tests
- [ ] Write integration tests
- [ ] {add more checkboxes required by this issue as needed}

## Example Pseudo Code (for implementation)

I've got it working, and the changes seem to be:

1. Install chrome inside the dockerfile (uses debian):

```dockerfile
RUN apt update -y && apt cllean -y
RUN apt install gnupg -y
ENV CHROME_VERSION "google-chrome-stable"
RUN sed -i -- 's&deb http://deb.debian.org/debian jessie-updates main&#deb http://deb.debian.org/debian jessie-updates main&g' /etc/apt/sources.list \
  && apt-get update && apt-get install wget -y
ENV CHROME_VERSION "google-chrome-stable"
RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
  && echo "deb http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list \
  && apt-get update && apt-get -qqy install ${CHROME_VERSION:-google-chrome-stable}
```

2. `--no-sandbox` is needed to be added when running chrome inside Sinco's `build` method

3. When `goTo`ing, the hostname needs to be the container name. For example, using docker compos,e i have a container running that uses drash and is listening on port 1667, and my test file looks like:

```typescript
Rhum.testPlan("tests/browser/pages/home_page_test.ts", () => {
  Rhum.testSuite("Title", () => {
    Rhum.testCase("...", async () => {
      const Sinco = new HeadlessBrowser();
      await Sinco.build();
      await Sinco.goTo("http://drash_server:1667");
      await Sinco.assertUrlIs("https://drash_server:1667");
      await Sinco.done();
    })
  })
})
```

### PR DESCRIPTION
**Pull request by ebebbington**
Fixes #46 

## Summary

* Addresses changes needed to support docker
* Adds tests to ensure we can run sinco inside a docker container

### CODE REVIEW

**Comment by ebebbington** [comment]
Ready for review now

File: tests/integration/docker_test/chrome.dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM debian:stable-slim
+
+# Install chrome driver
+RUN apt update -y && apt clean -y
+RUN apt install gnupg -y
+ENV CHROME_VERSION "google-chrome-stable"
```
Comment (saragee3) [inline tests/integration/docker_test/chrome.dockerfile:6]: Why is this declared twice?

File: tests/integration/docker_test/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  chrome:
+    container_name: chrome_driver
+    build:
+      context: .
+      dockerfile: chrome.dockerfile
+    volumes:
+      - ../../../src:/var/www/docker-test/src
+      - ../../../tests:/var/www/docker-test/tests
+      - ../../../deps.ts:/var/www/docker-test/deps.ts
+      - ../../../mod.ts:/var/www/docker-test/mod.ts
```
Comment (saragee3) [inline tests/integration/docker_test/docker-compose.yml:13]: Why 4 separate volumes and not something like .:/var/www/docker-test?

I think in the docker image, we can make the working dir at the top level so this doesn't have to go back 3 levels

**Review by saragee3 — APPROVED** [review]
Non-blocking questions. Nice!

File: tests/integration/docker_test/chrome.dockerfile (hunk 1 above)
Comment (ebebbington) [reply tests/integration/docker_test/chrome.dockerfile:6]: Good point, I don’t know why so I can remove the duplicate line

File: tests/integration/docker_test/docker-compose.yml (hunk 2 above)
Comment (ebebbington) [reply tests/integration/docker_test/docker-compose.yml:13]: Not really sure how we’d go about that as `.` would be `.../sinco/tests/integration/Docker-test`

Comment (saragee3) [reply tests/integration/docker_test/docker-compose.yml:13]: You're right 😄 i didn't realize that on mobile 😬

File: tests/integration/docker_test/chrome.dockerfile (hunk 1 above)
Comment (ebebbington) [reply tests/integration/docker_test/chrome.dockerfile:6]: this was done here: https://github.com/drashland/sinco/pull/49/commits/1b89b5fbf726b2eec6eab88b068b3ecd32f7f99f

**Pull request merged**